### ISSUE TITLE
Problems to install larcv

### ISSUE THREAD

**Issue by Gui1402** [DeepLearnPhysics/larcv#30]
Hi guys, 

I am beginner in Linux and python so I have some problems with larcv installation.

I was trying to install larcv following this link : [](http://deeplearnphysics.org/Blog/tutorials/tutorial-01.html)
My first attempt was using Linux Mint 19 
Then I cloned the repository, used `source configure.sh` and `make` but did not work.

Error:
`
Building core...

Compiling Base...
make[1]: Entering directory '/home/guilherme/larcv2/larcv/core/Base'
<< checking dependencies>>
<< generating dict >>
In file included from input_line_12:6:
In file included from /home/guilherme/larcv2/larcv/core/Base/ConfigManager.h:18:
In file included from /home/guilherme/larcv2/larcv/core/Base/larcv_base.h:19:
In file included from /home/guilherme/larcv2/larcv/core/Base/larcv_logger.h:21:
In file included from /home/guilherme/larcv2/larcv/core/Base/LArCVTypes.h:22:
In file included from /home/guilherme/anaconda3/lib/gcc/x86_64-unknown-linux-gnu/4.8.3/../../../../include/c++/4.8.3/climits:41:
In file included from /home/guilherme/anaconda3/include/limits.h:125:
In file included from /home/guilherme/anaconda3/etc/cling/lib/clang/3.7.0/include/limits.h:37:
/usr/include/limits.h:145:17: error: token is not a valid binary operator in a
      preprocessor subexpression
#if __GLIBC_USE (IEC_60559_BFP_EXT)
    ~~~~~~~~~~~ ^
Error: rootcling: compilation failure (/home/guilherme/larcv2/build/lib/libLArCVBase784ddeac21_dictUmbrella.h)
/home/guilherme/larcv2/Makefile/GNUmakefile.CORE:41: recipe for target '/home/guilherme/larcv2/build/Base/BaseDict.cxx' failed
make[1]: *** [/home/guilherme/larcv2/build/Base/BaseDict.cxx] Error 1
make[1]: Leaving directory '/home/guilherme/larcv2/larcv/core/Base'
GNUmakefile:34: recipe for target 'obj' failed
make: *** [obj] Error 2
`

So my next attempt was to use Ubuntu 16.04 and did not have problems to install larcv but when I import larcv on Jupyter(or spyder) the kernel crashed.

Did someone have the same problem?

**Comment by drinkingkazu** [comment]
A few comments:

0) I am happy to help debugging. Shall we focus first on Ubuntu? Would you share the error message (which shows up on the terminal that runs jupyter notebook, not on notebook itself)?

1) Have you considered using our singularity image? We use singularity software container to address a few issues including reproducibility of analysis and distribution of a software. What's related to your issue is that a container is designed to deliver you a working larcv development (as well as analysis) environment without having you install dependent libraries.

I use singularity on my laptop, a GPU server, and even on a GPU cluster / cloud. If you are interested in but don't know what it is, here's [introduction tutorial](https://github.com/DeepLearnPhysics/playground-singularity/wiki) and our singularity [recipe repository](https://github.com/DeepLearnPhysics/larcv2-singularity) as well as [built image repository](https://singularity-hub.org/collections/459).

**Comment by coreyjadams** [comment]
Do you need a custom build of larcv? Every if you do, singularity might be
the answer. You can solve a lot of the dependancies that way pretty easily.

Check out the larcv singularity builds and see if that works. The wiki page
by Kazu gives great instructions.

**Comment by coreyjadams** [comment]
Yeah I echo Kazu. I use singularity on my ubuntu laptop to run an ubuntu
image with root, tensorflow and larcv.  It's just easier to make a nice
consistent environment that I know will work elsewhere too.

**Comment by Gui1402** [comment]
Thank you for reply guys

Really I never heard about singularity image, maybe it can help me because I would like to run this example: 

http://deeplearnphysics.org/Blog/tutorials/tutorial-02.html

Do I need to install VirtualBox(or similar programs) to run a singularity image?


But when I after put `from larcv import larcv` on Jupyter notebook, the jupyter's terminal shows:
`*** Break *** bus error
[I 23:30:07.095 NotebookApp] KernelRestarter: restarting kernel (1/5), keep random ports
kernel 431c86fd-3727-43cb-8bd3-f3cabb344225 restarted`

**Comment by coreyjadams** [comment]
You should check out the singularity site for information on how to run it: http://singularity.lbl.gov/docs-run

And, Kazu's wiki page for the larch singularity in detail:
https://github.com/DeepLearnPhysics/playground-singularity/wiki
